cypress/integration/parallel-2/braze.spec.js

```diff
@@ -0,0 +1,135 @@
+
+const handleGuCookieError = () => {
+	cy.on('uncaught:exception', (err, runnable) => {
+		// When we set the `GU_U` cookie this is causing the commercial bundle to try and do
+		// something with the url which is failing in Cypress with a malformed URI error
+		expect(err.message).to.include('URI malformed');
+		// This error is unrelated to the test in question so return  false to prevent
+		// this commercial error from failing this test
+		return false;
+	});
+};
+
+describe('Braze messaging', function () {
+	beforeEach(function () {
+		cy.clearLocalStorage();
+		handleGuCookieError();
+		setLocalBaseUrl();
+	});
+
+	const visitArticle = (
+		url = 'https://m.code.dev-theguardian.com/games/2018/aug/23/nier-automata-yoko-taro-interview',
+	) => {
+		return cy.visit(`/Article?url=${url}`);
+	};
+
+	const acceptConsents = () => {
+		// Set consents for two geo locations, Europe which is typical for local
+		// dev and Team City and USA for GitHub actions
+		// Europe
+		cy.setCookie('consentUUID', 'dd166371-fd7f-4294-a094-467e44613f61');
+		// USA
+		cy.setCookie('ccpaConsentAll', 'true');
+		cy.setCookie('consentStatus', 'consentedAll');
```

Comment:
Thinking about this a little more, it seems like the path forward is just to stub [`window.guCmpHotFix.getConsentFor`][1], like so:

```js
window.guCmpHotFix.getConsentFor = (vendor, state) => {
  if(vendor === "braze") return true;
  return false;
};
```

[1]: https://github.com/guardian/consent-management-platform/blob/c576f20c306ef976d2e085ae93729c75fc8b6ebf/src/index.ts#L134

Reply:
Good point! I just looked at this with Max and I'm going to try an alternative approach here which should be less brittle, and also agnostic about the geo location.

Reply:
In the end I settled on interacting with the CMP banner for privacy settings, just like a real user would. I'm forcing the geolocation so that this can work consistently across dev and CI (which typically runs on servers in the US).

Reply:
Ah cool, so does that mean we no longer have the problem of the tests failing each time the cookie expires?

Reply:
Yeah exactly. This is now working the same way a user would give their consents, using the banner. I did go down this route initially but found things broken in CI as the GH action runs on US servers (I think that's what the issue was). But now I'm forcing the geolocation to `GB` for consistent behaviour regardless of environment.